### PR TITLE
Refactor devices routes to be on a separate file

### PR DESCRIPTION
**Pull request by biancarosa**
Yesterday while coding the update playbook method I noticed some improvements and talked to @acosferreira about it today.

I want to do some unit tests for the code pushed yesterday, so this should make things easier to test too.

- Creates devices route file.
- Separates device and update context
- Makes getUpdate work again
- Delete code that is not used or could be reused

### CODE REVIEW

File: pkg/routes/devices.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	// var devices []models.Device
+	var results []models.Device
+	//var results []models.UpdateTransaction
+	account, err := common.GetAccount(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	uuid := chi.URLParam(r, "DeviceUUID")
+	result := db.DB.
```
Comment (acosferreira) [inline pkg/routes/devices.go:64]: Is not possible use Gorm to do this query/relation?

File: pkg/routes/devices.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	DeviceUUID string
+	Tag        string
+}
+
+//  DeviceCtx is a handler for Device requests
+func DeviceCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var uCtx DeviceContext
+		uCtx.DeviceUUID = chi.URLParam(r, "DeviceUUID")
+		uCtx.Tag = chi.URLParam(r, "Tag")
```
Comment (acosferreira) [inline pkg/routes/devices.go:42]: we're handling the tag, but we're not using, what do you think to let is commented until we start to work with the tags?

File: pkg/routes/devices.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	})
+}
+
+// GetDeviceStatus returns the device with the given UUID that is associate to the account.
+// This is being used for the inventory table to determine whether the current device image
+// is the latest or older version.
+func GetDeviceStatus(w http.ResponseWriter, r *http.Request) {
+	// var devices []models.Device
+	var results []models.Device
+	//var results []models.UpdateTransaction
```
Comment (acosferreira) [inline pkg/routes/devices.go:57]: what do you think about remove this commented var to keep the code clean?

File: pkg/routes/devices.go (hunk 1 above)
Comment (biancarosa) [reply pkg/routes/devices.go:64]: I'm not sure but I'd rather not mess with this one since I just moved the query to a different function

**Review by acosferreira — APPROVED** [review]
LGTM

**Pull request merged**